### PR TITLE
add layerwise learning rate for adamw

### PR DESCRIPTION
**Pull request by zhaoyinglia**
### PR types

 New features
### PR changes

OPs
### Describe

add layerwise learningrate feature for adamw

### CODE REVIEW

File: paddle/fluid/operators/optimizers/adam_op.cc [hunk 1]
```diff
@@ -236,6 +236,10 @@ class AdamWOpMaker : public AdamOpMaker {
  public:
   void Make() {
     AdamOpMaker::Make();
+    AddAttr<float>("lr_ratio",
```
Comment (JZ-LIANG) [inline paddle/fluid/operators/optimizers/adam_op.cc:239]: why add this argument to adam? is that adamw and adam share the same .cc file ?

in this case, adamw should have its own .cc file

File: python/paddle/optimizer/adamw.py [hunk 2]
```diff
@@ -163,6 +165,9 @@ def __init__(self,
         self._apply_decay_param_fun = apply_decay_param_fun
         self._coeff = coeff
         self._lr_to_coeff = dict()
+        if lr_ratio is not None:
+            assert isinstance(lr_ratio, Callable)
+        self._lr_ratio = lr_ratio
```
Comment (JZ-LIANG) [inline python/paddle/optimizer/adamw.py:177]: you should think about how many kernel will be affected by  "lr_ratio". 
if you only want the lr_ratio the affect gpu and cpu kernel, you should raise an Unimplement Error for xpu and npu here.

File: python/paddle/optimizer/adamw.py [hunk 3]
```diff
@@ -163,6 +165,9 @@ def __init__(self,
         self._apply_decay_param_fun = apply_decay_param_fun
         self._coeff = coeff
         self._lr_to_coeff = dict()
+        if lr_ratio is not None:
```
Comment (JZ-LIANG) [inline python/paddle/optimizer/adamw.py:172]: should add explanation for the new lr_ration argument, which should follow the explanation for "apply_decay_param_fun"

File: paddle/fluid/operators/optimizers/adam_op.cc (hunk 1 above)
Comment (zhaoyinglia) [reply paddle/fluid/operators/optimizers/adam_op.cc:239]: AdamWOpMaker inherits AdamOpMaker, and they use the same InferShape function of AdamOp.
In this case, 'lr_ratio' has no effect on Adam.

File: python/paddle/optimizer/adamw.py (hunk 2 above)
Comment (zhaoyinglia) [reply python/paddle/optimizer/adamw.py:177]: done.

File: python/paddle/optimizer/adamw.py (hunk 3 above)
Comment (zhaoyinglia) [reply python/paddle/optimizer/adamw.py:172]: done.

**Review by PangHua — DISMISSED** [review]

**Review by PangHua — APPROVED** [review]

**Review by JZ-LIANG — APPROVED** [review]
LGTM

**Pull request merged**